napari/_qt/layer_controls/qt_layer_controls_base.py

```diff
@@ -157,6 +157,9 @@ def __init__(self, layer: Layer) -> None:
         self.opacityLabel.setEnabled(
             self.layer.blending not in NO_OPACITY_BLENDING_MODES
         )
+        if self.__class__ == QtLayerControls:
+            # this if is to prevent leaking of buttons in the tests
+            self.layout().addRow(self.button_grid)
```

Comment:
As above — can you explain why things were leaking, and how this fixes that issue?

Reply:
response above